demo-site/servers/nodejs/app.js

```diff
@@ -0,0 +1,202 @@
+    app.use(PxMiddlewareWrap);
+
+    if (pxInstance.cdEnforcer) {
+        PxCdMiddleware = pxInstance.cdMiddleware;
+        PxCdInterval = pxInstance.cdEnforcer.setIntervalId;
+        app.use(PxCdMiddleware);
+    }
+    app.use((req, res, next) => {
+        for (const [name, value] of Object.entries(req.headers)) {
+            res.setHeader(name, value);
+        }
+        next();
+    });
+}
+
+const initializeConfigs = () => {
+    return addEnvConfigs(pxConfigJson);
+}
+
+const setAdditionalActivityHandler = (pxConfig) => {
+    pxConfig['px_additional_activity_handler'] = (pxCtx) => {
+        const { uri, pxde, pxdeVerified, score } = pxCtx;
+        axios.post(pxConfig.px_backend_url + "/additional" + uri, {
+            _pxde: pxde,
+            pxdeVerified: pxdeVerified,
+            pxScore: score
+        }).catch((e) => console.log(e.message));
+    };
+}
+
+const setCustomParam = (pxConfig) => {
+    pxConfig['px_enrich_custom_parameters'] = (px_context, px_config)=>{
+        let customParams = [];
+        for (let i = 1; i < 3; i++) {
+            let param_key = `custom_param${i}`;
+            let value = `test${i}`;
+            customParams[param_key] = value;
+        }
+        for (let i = 3; i < 7; i++) {
+            let param_key = `custom_param${i}`;
+            let value = i;
+            customParams[param_key] = value;
+        }
+        for (let i = 7; i <= 12; i++) {
+            let param_key = `custom_param${i}`;
+            let value = null;
+            customParams[param_key] = value;
+        }
+        return customParams;
+    };
+}
+
+const setTestEndpoints = (app) => {
+    app.post('/config', function (req, res, next) {
+        if(process.env.ENABLE_TEST_ENDPOINTS === 'false'){
+            return res.sendStatus(404);
+        }
+        let newConfig = req.body;
+        // merge new config into pxConfig
+        Object.assign(pxConfig, newConfig)
+        if (pxConfig['px_csp_enabled']) {
+            clearInterval(PxCdInterval);
+        }
+        var pxInstance = perimeterx.new(pxConfig);
+        PxMiddleware = pxInstance.middleware;
+        setAdditionalActivityHandler(pxConfig);
+        setCustomParam(pxConfig)
+        res.sendStatus(200);
+    });
+    
+    app.get('/supported-features', function (req, res, next) {
+        if(process.env.ENABLE_TEST_ENDPOINTS === 'false'){
+            return res.sendStatus(404);
+        }
+        const supportedFeatures = require('perimeterx-node-express/px_metadata.json');
+        return res.json(supportedFeatures);
+    });
+    
+    app.get('/test-app-credentials', function (req, res){
+        if (process.env.ENABLE_TEST_ENDPOINTS === 'false'){
+            return res.sendStatus(404);
+        }
+        const test_app_credentials = {
+            "px_app_id": pxConfig.px_app_id,
+            "px_cookie_secret": pxConfig.px_cookie_secret
+        };
+        return res.json(test_app_credentials)
+    });
```

Comment:
Are these test endpoints still necessary? I'm assuming yes but just checking to make sure we don't have unnecessary code.